### PR TITLE
Fix/adjacency and friction

### PR DESCRIPTION
**Pull request by cremebrule**

### CODE REVIEW

**Review by ChengshuLi — APPROVED** [review]
LGTM, with minor comments!

File: omnigibson/prims/joint_prim.py [hunk 1]
```diff
@@ -433,7 +434,9 @@ def friction(self, friction):
         Args:
             friction (float): friction to set
         """
-        self._articulation_view.set_friction_coefficients(np.array([[friction]]), joint_indices=self.dof_indices)
+        self.set_attribute("physxJoint:jointFriction", friction)
```
[the remark this answers is not in the document]
Comment (cremebrule) [reply omnigibson/prims/joint_prim.py:436]: yes

File: omnigibson/prims/cloth_prim.py [hunk 2]
```diff
@@ -243,6 +250,16 @@ def keypoint_particle_positions(self):
...
 
+    @property
+    def centroid_particle_position(self):
+        """
+        Grabs the individual particle that was pre-computed to be the closest to the centroid of this cloth prim.
+
+        Returns:
+            np.array: centroid particle's (x,y,z) cartesian coordinates relative to the world frame
+        """
+        return self.compute_particle_positions(idxs=[self._centroid_idx])[0]
```
[the remark this answers is not in the document]
Comment (cremebrule) [reply omnigibson/prims/cloth_prim.py:261]: it was originally to make it clear that there's non-trivial compute required in this call, so it should be used sparingly if possible

File: omnigibson/object_states/adjacency.py [hunk 3]
```diff
@@ -80,17 +86,48 @@ def compute_adjacencies(obj, axes, max_distance):
...
+        if not use_aabb_center:
+            # Dynamically compute start points by iterating over the directions and pre-shooting rays from
+            # which to shoot back from
+            # For a given direction, we go in the negative (opposite) direction to the edge of the object extent,
+            # and then proceed with an additional offset before shooting rays
+            shooting_offset = 0.01
+
+            direction_extent = directions * (aabb_higher - aabb_lower).reshape(1, 3) / 2.0
+            pre_start = object_position.reshape(1, 3) - (direction_extent + np.ones((1, 3)) * shooting_offset)
+            pre_end = object_position.reshape(1, 3) + direction_extent
```
[the remark this answers is not in the document]
Comment (cremebrule) [reply omnigibson/object_states/adjacency.py:106]: right, but the direction extent can be negative or positive, right?

**Pull request merged**